### PR TITLE
Feature/#166 request events

### PR DESCRIPTION
**Pull request by denwilliams**
hook up to travis, and add events for request lifecycle

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -85,6 +85,57 @@ closeServices
...
+  client.on('request:error', doStuff);
+});
+```
+
+### request:start
+
+```js
+client.on('request:start', function(e) {
+  console.log(e.method); // the method name eg "getCustomers" (string)
+  console.log(e.request); // the full request object
```
Comment (aantthony) [inline README.md]: What properties should be on the request object?

Comment (denwilliams) [inline README.md]: hmm... i'll have to try and reverse engineer it... not documented anywhere at the moment

Comment (denwilliams) [inline README.md]: added doc for request and response

File: lib/communications.js [hunk 2]
```diff
@@ -21,6 +23,8 @@ function Communications (url, logger) {
   this.serviceMetaRepliers = {};
 }
 
+util.inherits(Communications, EventEmitter);
```
Comment (aantthony) [inline lib/communications.js]: You should also call the super method from the constructor:

```js
EventEmitter.call(this);
```

**Pull request merged**